include/nbl/builtin/hlsl/math/functions.hlsl

```diff
@@ -0,0 +1,453 @@
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_MATH_FUNCTIONS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_MATH_FUNCTIONS_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/numbers.hlsl"
+#include "nbl/builtin/hlsl/spirv_intrinsics/core.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace math
+{
+
+namespace impl
+{
+template<typename T, uint32_t LP, bool Odd=(LP&0x1)>
+struct lp_norm;
+
+// infinity case
+template<typename T>
+struct lp_norm<T,0,false>
```

Comment:
you'll never hit it because 0 is not Odd :)

Reply:
`0 & 0x1` doesn't evaluate to `false`?

Reply:
it does

Reply:
my bad, excuse my braindeadness.

Reply:
happens